### ISSUE TITLE
Not detecting REAL tag

### ISSUE THREAD

**Issue by duramato** [guessit-io/guessit#408]
```
For: Pawn.Stars.S12E20.The.Pawn.Awakens.REAL.READ.NFO.720p.HDTV.x264-DHD
GuessIt found: {
    "title": "Pawn Stars", 
    "season": 12, 
    "episode": 20, 
    "episode_title": "The Pawn Awakens REAL READ NFO", 
    "screen_size": "720p", 
    "format": "HDTV", 
    "video_codec": "h264", 
    "release_group": "DHD", 
    "type": "episode"
}
```

```
For: Pawn.Stars.S12E22.Racing.Revolution.REAL.720p.HDTV.x264-DHD
GuessIt found: {
    "title": "Pawn Stars", 
    "season": 12, 
    "episode": 22, 
    "episode_title": "Racing Revolution REAL", 
    "screen_size": "720p", 
    "format": "HDTV", 
    "video_codec": "h264", 
    "release_group": "DHD", 
    "type": "episode"
}
```

**Comment by ratoaq2** [comment]
The main issue is that we can't rely on rules that depends on whether a word is all uppercase or not.

The REAL detection depends whether there are matches surrounding it. For both cases there's no previous match (episode_title is not counted since it's derived from these empty spaces)

For instance:
`Ridiculousness.S04E08.REAL.HDTV.x264-YesTV.[GloTV]`

REAL has an episode and video_codec as neighbors.
`Pawn.Stars.S12E22.Racing.Revolution.REAL.720p.HDTV.x264-DHD`
has only screen_size as neighbor.

The following example has the same pattern, but REAL should be part of episode_title:
`Outrageous.Acts.of.Science.S05E02.Is.This.for.Real.720p.HDTV.x264-DHD`

Using natural language, can you describe a rule to detect REAL that doesn't rely on case sensitivity?

**Comment by fernandog** [comment]
@duramato what is `.READ.NFO.` ?

maybe that matches a kind tag (which is always before resolution), i think is ok to match REAL before it. Agree @ratoaq2 ?

**Comment by Toilal** [comment]
I think read nfo should be a new possible value for other property

**Comment by ratoaq2** [comment]
I'm already handling Read NFO in my local branch. I'm revisiting #302 

One of the valid points to discuss is the `case` information. I don't know if I agree or disagree with completely ignoring it. If user provides an input string like that, that's already a hint that guessit could use.

**Comment by duramato** [comment]
READ.NFO  isnt a "proper" tag

**Comment by duramato** [comment]
A rule should be if it is all in caps in a proper tag, if everything is in lower case, check if the word preceding the resolution is REAL

**Comment by VeNoMouS** [comment]
Well... no... because what if the episode title is all in CAPS... and "READ.NFO" is not a scene standard tag ... write a rule yourself... data['title'].replace['READ.NFO,', ''] 

![image](https://cloud.githubusercontent.com/assets/597545/23442631/95ef1914-fe8e-11e6-8e31-eb5b4c0da0f4.png)

**Comment by ratoaq2** [comment]
The main problem in this issue is the REAL tag. 
READ.NFO is just the next word in the first example, which doesn't follow exactly scene standards, but guessit doesn't guess only scene releases. 

The question remains: how to know REAL is a scene tag and not just part of one of the titles. 
I've seen filenames where is all lowercase and you can't distinguish it 100%

**Comment by Toilal** [comment]
Yes, i agree, it seems we can't fix this one because I don't want guessit
to rely on characters case.

**Comment by duramato** [comment]
> The question remains: how to know REAL is a scene tag and not just part of one of the titles.


This isn't any different than an episode title ending with the word proper, repack or rerip... so guessit's behaviour should be the same.

**Comment by ratoaq2** [comment]
Real is a common english word. Different from RERIP, REPACK or PROPER:

`Outrageous.Acts.of.Science.S05E02.Is.This.for.Real.720p.HDTV.x264-DHD`
That shouldn't be REAL, since its episode_title is `Is This for Real`

**Comment by duramato** [comment]
```
For: Cleveland.Hustles.S01E03.Downward.Dogs.and.Proper.Pigs.720p.HDTV.x264-W4F
GuessIt found: {
    "title": "Cleveland Hustles", 
    "season": 1, 
    "episode": 3, 
    "episode_title": "Downward Dogs and", 
    "other": "Proper", 
    "proper_count": 1, 
    "screen_size": "720p", 
    "format": "HDTV", 
    "video_codec": "h264", 
    "release_group": "W4F", 
    "type": "episode"
}
```

**Comment by duramato** [comment]
`Outrageous.Acts.of.Science.S05E02.Is.This.for.Real.720p.HDTV.x264-DHD`
In that example the _proper_ tag isn't proceeded by numbering and followed by resolution. It is preceeded by:  `Is.This.for`

**Comment by VeNoMouS** [comment]
@duramato tbh, that is tagged wrong..

![image](https://user-images.githubusercontent.com/597545/36950731-48976588-205f-11e8-85af-4a52f0c59244.png)


scene rules state

```                                                                           ³
³ 15) [ Tagging ]                                                             ³
³  15.1) Only the following additional tags are allowed:                      ³
³         ALTERNATIVE.CUT, CONVERT, COLORIZED, DC, DIRFIX, DUBBED, EXTENDED,  ³
³         FINAL, INTERNAL, NFOFIX, OAR, OM, PPV, PROPER, REAL, REMASTERED,    ³
³         READNFO, REPACK, RERIP, SAMPLEFIX, SOURCE.SAMPLE, SUBBED,           ³
³         UNCENSORED, UNRATED, UNCUT, WEST.FEED, and WS.                      ³
³   15.1.1) WEST.FEED refers to an alternative version which airs             ³
³           exclusively on the west coast, such as a live episode of          ³
³           Undateable which has two separate performances of the same        ³
³           episode for each coast, east and west.                            ³
³    15.1.1.1) Releases must be tagged as WEST.FEED when they come from an    ³
³              exclusive west coast airing, even if no east feed has been     ³
³              released first.                                                ³
³  15.2) Variations of any additional tags are not allowed.                   ³
³         e.g. READ.NFO or RNFO is not allowed, READNFO must be used.         ³
³  15.3) READNFO should be used sparingly. Discretion is recommended.         ³
³   15.3.1) The READNFO tag must not be used with PROPER, REPACK or RERIP.    ³
³           The NFO is required to contain a reason, therefore the tag is     ³
³           redundant.                                                        ³
³  15.4) Tags must only be used once, but the order is left to the            ³
³        discretion of the group.                                             ³
³   15.4.1) Except in situations where the REAL tag is required to be         ³
³           stacked to differentiate between multiple invalid releases.       ³
³            e.g. A REAL.REAL.PROPER is required for a REAL.PROPER and        ³
³                 PROPER.  
```

there it would look more like 

![image](https://user-images.githubusercontent.com/597545/36950739-4e6e1d08-205f-11e8-8ee5-a5488a7d1c83.png)

There is no PROPER tag in there...

**Comment by VeNoMouS** [comment]
Actually that said, I didnt even follow scene rules... , scene rules state REAL.PROPER, if we do REAL.PROPER instead of PROPER.REAL

*REAL* becomes part of the episode title... so there def is an issue here in regards to the PROPER parentheses..

![image](https://user-images.githubusercontent.com/597545/36950821-6551bef2-2060-11e8-9837-adfe2cf037f9.png)

**Comment by duramato** [comment]
All REAL issues are fixed, guessit handles those just fine now. 
The issue is it detects episode titles that have a _proper_ tag in them like the example I gave.

```
For: Vampirina.S01E16.REAL.HDTV.x264-W4F
GuessIt found: {
    "title": "Vampirina",
    "season": 1,
    "episode": 16,
    "other": "Proper",
    "proper_count": 1,
    "source": "HDTV",
    "video_codec": "H.264",
    "release_group": "W4F",
    "type": "episode"
}
```
```
For: Outrageous.Acts.of.Science.S05E02.Is.This.for.Real.720p.HDTV.x264-DHD
GuessIt found: {
    "title": "Outrageous Acts of Science", 
    "season": 5, 
    "episode": 2, 
    "episode_title": "Is This for Real", 
    "screen_size": "720p", 
    "source": "HDTV", 
    "video_codec": "H.264", 
    "release_group": "DHD", 
    "type": "episode"
}
```

While it detects `Cleveland.Hustles.S01E03.Downward.Dogs.and.Proper.Pigs.720p.HDTV.x264-W4F` as having a _proper_ tag, which it doesn't

````
For: Cleveland.Hustles.S01E03.Downward.Dogs.and.Proper.Pigs.720p.HDTV.x264-W4F
GuessIt found: {
    "title": "Cleveland Hustles",
    "season": 1,
    "episode": 3,
    "episode_title": "Downward Dogs and",
    "other": "Proper",
    "proper_count": 1,
    "screen_size": "720p",
    "source": "HDTV",
    "video_codec": "H.264",
    "release_group": "W4F",
    "type": "episode"
}
```

**Comment by ratoaq2** [comment]
I'm revisiting this issue.

The scenario with an issue with `Proper` can be easily fixed with no side effect.

About Real, this is my opinion:
1. Wrong `title` is a big problem, so we should not break that.
2. Not detecting `Real` (Proper) is a medium problem
3. Wrong `episode_title` detection is a small problem.

Probably the `Real` detection could be enhanced in a way that it will never cause problems with `title`, but it could cause a problem with `episode_title` in very rare cases. Having a solution that keep 1, fixes 2 and 3 only happen in very rare cases. From my point of view that's a good improvement.

**Comment by ratoaq2** [comment]
I'm pushed the changes to fix this issue: 53e8e8626e9b6cfd86251a05e2f07f35bb32dada

We can revisit it if needed
